src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -656,14 +657,30 @@ internal void Log(GrpcEvent msg)
                 // Restore the execution context from the original invocation. This allows AsyncLocal state to flow to loggers.
                 System.Threading.ExecutionContext.Run(context.AsyncExecutionContext, (s) =>
                 {
-                    if (rpcLog.Exception != null)
+                    if (rpcLog.LogCategory == RpcLogCategory.CustomMetric)
                     {
-                        var exception = new Workers.Rpc.RpcException(rpcLog.Message, rpcLog.Exception.Message, rpcLog.Exception.StackTrace);
-                        context.Logger.Log(logLevel, new EventId(0, rpcLog.EventId), rpcLog.Message, exception, (state, exc) => state);
+                        var rpcLogProperties = rpcLog.PropertiesMap.ToDictionary(i => i.Key, i => i.Value);
```

Comment:
Thinking about performance here -- we turn this into a dictionary here, then later pull out two of the values and create another dictionary.

Can we instead exclude the name and value here and just create this dictionary once? Then you can pull the metric name and values straight from PropertiesMap?

Reply:
great point, fixing!